test/dummy_solver.jl

```diff
@@ -12,38 +12,57 @@ function dummy_solver(nlp :: AbstractNLPModel;
   nvar, ncon = nlp.meta.nvar, nlp.meta.ncon
   T = eltype(x)
 
-  fx = obj(nlp, x)
-  cx = ncon > 0 ? cons(nlp, x) : zeros(T)
-  α = one(T)
+  cx = ncon > 0 ? cons(nlp, x) : zeros(T, 0)
+  gx = grad(nlp, x)
+  Jx = ncon > 0 ? jac(nlp, x) : zeros(T, 0, nvar)
+  y = -Jx' \ gx
+  Hxy = ncon > 0 ? hess(nlp, x, y) : hess(nlp, x)
+
+  dual = gx + Jx' * y
 
   iter = 0
-  @info log_header([:iter, :f, :c, :t], [Int, T, T, Float64])
-  @info log_row(Any[iter, fx, norm(cx), elapsed_time])
+
+  ϵd = atol + rtol * norm(dual)
+  ϵp = atol
+
+  fx = obj(nlp, x)
+  @info log_header([:iter, :f, :c, :dual, :t], [Int, T, T, Float64])
+  @info log_row(Any[iter, fx, norm(cx), norm(dual), elapsed_time])
+  solved = norm(dual) < ϵd && norm(cx) < ϵp
   tired = neval_obj(nlp) + neval_cons(nlp) > max_eval || elapsed_time > max_time
 
-  while !tired
-    xt = x + α * convert.(T, randn(nvar))
-    ft = obj(nlp, xt)
-    ct = ncon > 0 ? cons(nlp, xt) : zeros(T)
-    if ft < fx && norm(ct) ≤ norm(cx)
-      x .= xt
-      fx = ft
-      cx .= ct
-    else
-      α = 9α / 10
-    end
+  while !(solved || tired)
+    Hxy = ncon > 0 ? hess(nlp, x, y) : hess(nlp, x)
+    W = [Symmetric(Hxy, :L)  Jx'; Jx  zeros(T, ncon, ncon)]
```

Comment:
Maybe you could construct a Symmetric() of the lower triangle `[Hxy spzeros(nvar, ncon) ; Jx spzeros(ncon, ncon)]`?!

Reply:
:+1:

Reply:
I've updated, but I'm not using sparse matrices because `lu` with sparse matrices of different precision is failing. I remember creating an issue related to this, but I can't find it.